contracts/src/interfaces/IHyperdriveMatchingEngine.sol

```diff
@@ -0,0 +1,168 @@
+        bytes32 indexed shortOrderHash,
+        address long,
+        address short
+    );
+
+    /// @notice The type of an order intent.
+    enum OrderType {
+        OpenLong,
+        OpenShort
+    }
+
+    /// @notice The order intent struct that encodes a trader's desire to trade.
+    struct OrderIntent {
+        /// @dev The trader address that will be charged when orders are matched.
+        address trader;
+        /// @dev The Hyperdrive address where the trade will be executed.
+        IHyperdrive hyperdrive;
+        /// @dev The amount to be used in the trade. In the case of `OpenLong`,
+        ///      this is the amount of base to deposit, and in the case of
+        ///      `OpenShort`, this is the amount of bonds to short.
+        uint256 amount;
+        /// @dev The slippage guard to be used in the trade. In the case of
+        ///      `OpenLong`, this is the minimum output in bonds, and in the
+        ///      case of `OpenShort`, this is the maximum deposit in base.
+        uint256 slippageGuard;
+        /// @dev The minimum vault share price. This protects traders against
+        ///      the sudden accrual of negative interest in a yield source.
+        uint256 minVaultSharePrice;
+        /// @dev The options that configure how the trade will be settled.
+        ///      `asBase` is required to be true, the `destination` is the
+        ///      address that receives the long or short position that is
+        ///      purchased, and the extra data is configured for the yield
+        ///      source that is being used.
+        IHyperdrive.Options options;
+        /// @dev The type of the order. This is either `OpenLong` or `OpenShort`.
+        OrderType orderType;
+        /// @dev The signature that demonstrates the source's intent to complete
+        ///      the trade.
+        bytes signature;
+        /// @dev The order's expiry timestamp. At or after this timestamp, the
+        ///      order can't be filled.
+        uint256 expiry;
+        /// @dev The order's salt. This introduces some randomness which ensures
+        ///      that duplicate orders don't collide.
+        bytes32 salt;
+    }
+
+    /// @notice Get the name of this matching engine.
+    /// @return The name string.
+    function name() external view returns (string memory);
+
+    /// @notice Get the kind of this matching engine.
+    /// @return The kind string.
+    function kind() external view returns (string memory);
+
+    /// @notice Get the version of this matching engine.
+    /// @return The version string.
+    function version() external view returns (string memory);
+
+    /// @notice Get the Morpho flash loan provider.
+    /// @return The Morpho contract address.
+    function morpho() external view returns (IMorpho);
+
+    /// @notice Check if an order has been cancelled.
+    /// @param orderHash The hash of the order to check.
+    /// @return True if the order was cancelled.
```

Comment:
Assuming false if otherwise? What if the `orderHash` wasn't a valid `orderHash`? Still false?

Reply:
I'll add something about what false means.

What would make an order hash invalid? This is just a mapping from `bytes32` to `bool`. In theory, there are orders that hash to every `bytes32` value, we just don't know what they are because of the hash function's pre-image resistance.

Reply:
When I wrote this I hadn't read the implementation yet, totally get you on the hash-validity front